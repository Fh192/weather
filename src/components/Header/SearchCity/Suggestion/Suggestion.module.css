.suggestion {
  font-weight: 600;
  font-size: 15px;
  cursor: pointer;
  padding: 7px 20px;
}

.suggestion:hover {
  background: #eceef2;
}
