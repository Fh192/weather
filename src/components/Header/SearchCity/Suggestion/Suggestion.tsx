import React from 'react';
import s from './Suggestion.module.css';

interface Props {
  suggestion: string;
  onSelect: () => void;
}

export const Suggestion: React.FC<Props> = ({ suggestion, onSelect }) => {
  return (
    <li className={s.suggestion} onClick={onSelect}>
      {suggestion}
    </li>
  );
};
