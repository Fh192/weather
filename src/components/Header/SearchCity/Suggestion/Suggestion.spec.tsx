import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Suggestion } from './Suggestion';

describe('Suggestion.tsx', () => {
  it('renders correctly', async () => {
    const onSelect = jest.fn();
    const suggestion = 'Saint Petersburg';
    const { container } = render(
      <Suggestion suggestion={suggestion} onSelect={onSelect} />
    );

    expect(container).toMatchSnapshot();
  });
  it('should call "onSelect" on click', async () => {
    const onSelect = jest.fn();
    const suggestion = 'Saint Petersburg';
    render(<Suggestion suggestion={suggestion} onSelect={onSelect} />);

    userEvent.click(screen.getByText(suggestion));

    expect(onSelect).toBeCalledTimes(1);
  });
});
