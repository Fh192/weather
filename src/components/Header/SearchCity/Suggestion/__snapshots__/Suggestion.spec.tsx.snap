// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Suggestion.tsx renders correctly 1`] = `
<div>
  <li
    class="suggestion"
  >
    Saint Petersburg
  </li>
</div>
`;

exports[`Suggestion.tsx renders with "Saint Petersburg" suggestion`] = `
<div>
  <li
    class="suggestion"
  >
    Saint Petersburg
  </li>
</div>
`;
